jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveSqlGenerationVisitor.java

```diff
@@ -33,86 +31,94 @@ public HiveSqlGenerationVisitor(final SqlDialect dialect, final SqlGenerationCon
     @Override
     public String visit(final SqlSelectList selectList) throws AdapterException {
         final List<String> selectListElements = new ArrayList<>();
-        final SqlStatementSelect select = (SqlStatementSelect) selectList.getParent();
         if (selectList.isSelectStar()) {
-            if (SqlGenerationHelper.selectListRequiresCasts(selectList, nodeRequiresCast)) {
-                // Do as if the user has all columns in select list
-                int columnId = 0;
-                final List<TableMetadata> tableMetadata = new ArrayList<>();
-                SqlGenerationHelper.addMetadata(select.getFromClause(), tableMetadata );
-                for (final TableMetadata tableMeta : tableMetadata) {
-                    for (final ColumnMetadata columnMeta : tableMeta.getColumns()) {
-                        final SqlColumn sqlColumn = new SqlColumn(columnId, columnMeta);
-                        final String typeName = ColumnAdapterNotes.deserialize(sqlColumn.getMetadata().getAdapterNotes(), sqlColumn.getMetadata().getName()).getTypeName();
-                        if (typeName.equals("BINARY")) {
-                            selectListElements.add("base64(" + super.visit(sqlColumn) + ")");
-                        } else {
-                            selectListElements.add(super.visit(sqlColumn));
-                        }
-                        ++columnId;
-                    }
-                }
-            }
-            else {
-                selectListElements.add("*");
-            }
+            selectListElements.addAll(getSelectStarList(selectList));
         } else {
-            if(selectList.isRequestAnyColumn()){
+            if (selectList.isRequestAnyColumn()) {
                 return "1";
```

Comment:
Replace by `representAnyColumn()`

Reply:
Hive's logic is somehow different in this method. I couldn't fast find out a proper solution. I think we should just leave it until we refactor this part.

Reply:
`return SqlConstants.ONE;`

in that case.

Reply:
done